airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/oauth.py

```diff
@@ -126,3 +128,166 @@ def access_token(self) -> str:
     @access_token.setter
     def access_token(self, value: str):
         self._access_token = value
+
+
+@dataclass
+class DeclarativeSingleUseRefreshTokenOauth2Authenticator(AbstractOauth2Authenticator, DeclarativeAuthenticator):
```

Comment:
this feels like a lot of duplication -- shouldn't we use the existing single use authenticator and configure it using the user inputs instead?

Reply:
@sherifnada yeah I had the same thoughts but decided to duplicate because of two things:
 - thought we should probably have independent code bases for python & low code CDKs
 - reusing the existing single use authenticator would make it impossible to use interpolated strings as input parameters. 

On the other hand, I think we hardly need those fields to be interpolated.
Now that you have dispelled my doubts - I updated the code, please take a look again